File: code-ninja-course/JS/src/words/words.js
const words = [
    'dog', 'cat', 'horse', 'goat', 'cat', 'horse', 'fish', 'dog', 'dog', 'fish'  ];

function find() {

  var vector_sz = 0;
  var word;

  vector_sz = words.length;

  if (vector_sz === 0) {
    console.log('Please enter some words!');
    return -1;
  }

  var frequency = [];
  var unique_words = [];
  var count = 0;
  var isUnique;

  for (i = 0; i < words.length; i++) {
    isUnique = true;

    for (j = 0; j < unique_words.length; j++) {
      if (words[i] === unique_words[j]) {
        isUnique = false;
      }
    }

    if (isUnique) {
      unique_words.push(words[i]);
    }
  }
  for (i = 0; i < unique_words.length; i++) {
    for (j = 0; j < words.length; j++) {
      if (unique_words[i] == words[j]) {
        count = count + 1;
      }
    }
    frequency.push(count);
    count = 0;
  }

  for (i = 0; i < unique_words.length; i++) {
    console.log(unique_words[i] + ' ' + frequency[i]);
  }
  return 0;
}

File: code-ninja-course/C/src/words/words.c
#include <stdio.h>

int find()
{
	printf("Enter a list of words followed by end-of-file: ");

	char* words[10];
	char* uwords[10];
	char w[100];
	int i=0;

	for (int j = 0; j<10; j++)
	{
		words[j] = (char*) malloc(100);
		uwords[j] = (char*)malloc(100);
	}

	while (gets(w))
	{
		strcpy(words[i], w);
		i++;
	}


	if (i == 0)
	{
		printf("Please enter some words!");
		return -1;
	}

	int frequency[10];
	int findex = 0;
	int count = 0;
	int isUnique = 0;


	strcpy(uwords[0], words[0]);
	int uwords_sz = 1;

	for (unsigned k = 0; k < i; ++k)
	{
		isUnique = 1;

		for (unsigned j = 0; j < uwords_sz; ++j)
		{
			if (strcmp(words[k], uwords[j]) == 0)
			{
				isUnique = 0;
			}
		}
		if (isUnique)
		{
			strcpy(uwords[uwords_sz], words[k]);
			uwords_sz++;
		}
	}

	for (unsigned k = 0; k < uwords_sz; ++k)
	{
		for (unsigned j = 0; j < i; ++j)
		{
			if (strcmp(uwords[k], words[j]) == 0)
			{
				count += 1;
			}
		}
		frequency[findex] = count;
		findex++;
		count = 0;
	}

	for (unsigned m = 0; m < uwords_sz; ++m)
	{
		printf("%s      %i\n", uwords[m], frequency[m]);
	}
	return 0;
}
